tidy3d/components/lumped_element.py

```diff
@@ -81,22 +86,52 @@ class LumpedResistor(AbstractLumpedResistor, Box):
         "associated voltage drop will occur. Must be in the plane of the element.",
     )
 
-    _plane_validator = assert_plane()
+    snap_perimeter_to_grid: bool = pydantic.Field(
+        True,
+        title="Snap Perimeter to Grid",
+        description="When enabled, the perimeter of the lumped element is snapped to the simulation grid, "
+        "which improves accuracy when the number of grid cells is low within the element. Lumped elements "
+        "are always snapped to the grid along their normal axis.",
+    )
 
     @cached_property
     def normal_axis(self):
-        """Normal axis of the lumped element."""
+        """Normal axis of the lumped element, which is the axis where the element has zero size."""
         return self.size.index(0.0)
 
+    @cached_property
+    def lateral_axis(self):
+        """Lateral axis of the lumped element."""
+        return 3 - self.voltage_axis - self.normal_axis
+
+    @cached_property
+    def _voltage_axis_2d(self) -> Axis2D:
+        """Returns the voltage axis using the in-plane dimensions used by :class:`Medium2D`."""
+        if self.normal_axis > self.voltage_axis:
+            return self.voltage_axis
+        else:
+            return self.voltage_axis - 1
+
+    @cached_property
+    def _snapping_spec(self) -> SnappingSpec:
+        """Returns the snapping behavior for each dimension of the lumped element."""
+        snap_location = [SnapLocation.Boundary] * 3
+        snap_behavior = [SnapBehavior.Closest] * 3
+        snap_location[self.lateral_axis] = SnapLocation.Center
+        snap_behavior[self.lateral_axis] = SnapBehavior.Expand
```

Comment:
This was an option that improved the accuracy quite a bit. Generating the equivalent medium is quite sensitive to the exact size of the Box, and so I need to make sure that the "simulated" box is the same size as the box used to generated the equivalent medium. Basically, this snapping enforces that the component of electric field parallel to the current is fully enclosed by the equivalent media, so that staircasing will not change the dimensions of the box.